views/middleware.go

```diff
@@ -48,16 +51,43 @@ func (v *Views) RequiresPermission(h http.Handler, p permission.Permission) http
 			return
 		}
 
+		acceptedPerms := permission.GetValidPermissions(p)
+
 		for _, perm := range perms {
-			if perm.Name == p.Name {
+			if acceptedPerms[perm] {
 				h.ServeHTTP(w, r)
 				return
 			}
 		}
-		err = v.template.RenderNoNavsTemplate(w, nil, templates.ForbiddenTemplate)
+
+		err = v.template.RenderNoNavsTemplate(w, nil, templates.Forbidden500Template)
 		if err != nil {
 			log.Println(err)
 		}
 		w.WriteHeader(http.StatusForbidden)
 	}
 }
+
+func (v *Views) RequiresMinimumPermissionNoHttp(userID int, p permissions.Permissions) bool {
```

Comment:
Could `RequiresMinimumPermission` be rewritten in terms of `RequiresMinimumPermissionNoHttp`, to reduce duplication?

Reply:
They have been combined, both functions still exist but one has logic, the other calls it